data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetDeposits.java

```diff
@@ -0,0 +1,130 @@
+import tech.pegasys.teku.validator.coordinator.DepositProvider;
+
+@SuppressWarnings("unused")
+public class GetDeposits extends MigratingEndpointAdapter {
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/eth/v1/debug/beacon/deposits";
+  private static final String DEFAULT_STATE_ID = "head";
+
+  private static final SerializableTypeDefinition<List<Deposit>> DEPOSITS_RESPONSE_TYPE =
+      SerializableTypeDefinition.<List<Deposit>>object()
+          .name("GetDepositsResponse")
+          .withField(
+              "data", listOf(Deposit.SSZ_SCHEMA.getJsonTypeDefinition()), Function.identity())
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final DepositProvider depositProvider;
+  private final Spec spec;
+
+  public GetDeposits(
+      final DataProvider provider, final DepositProvider depositProvider, final Spec spec) {
+    this(provider.getChainDataProvider(), depositProvider, spec);
+  }
+
+  public GetDeposits(
+      final ChainDataProvider chainDataProvider,
+      final DepositProvider depositProvider,
+      final Spec spec) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getDeposits")
+            .summary("Deposits not yet included on the chain")
+            .description(
+                "Range of deposits that could be used in new block if we are assigned to be a proposer.")
+            .tags(TAG_DEBUG)
+            .response(SC_OK, "Request successful", DEPOSITS_RESPONSE_TYPE)
+            .build());
+    this.chainDataProvider = chainDataProvider;
+    this.depositProvider = depositProvider;
+    this.spec = spec;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Deposits not yet included on the chain",
+      description =
+          "Range of deposits that could be used in new block if we are assigned to be a proposer.",
+      tags = {TAG_DEBUG},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetDepositsResponse.class)),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    request.respondAsync(
+        chainDataProvider
+            .getBeaconState(DEFAULT_STATE_ID)
+            .thenApplyChecked(
+                maybeState -> {
+                  if (maybeState.isEmpty()) {
+                    return AsyncApiResponse.respondNotFound();
+                  } else {
+                    List<Deposit> allDeposits =
+                        depositProvider.getAllPendingDeposits(maybeState.get().getData());
+                    return AsyncApiResponse.respondOk(allDeposits);
```

Comment:
hmm, I don't think this is quite the Api we want.  There's three separate things this issue wants:

1. Report the range of deposits that are currently available. That wouldn't need a state input at all and should just return all the deposits we currently hold.
2. An API to return the Eth1Data that we'd include in a new block we propose. This would just always base it's decision on the current head state because our Eth1 block cache will have been pruned for earlier states and so won't give the same answer it would have at the time.
3. An API that takes a state ID and returns the Eth1Data from the state, plus the bits of calculated data from that state mentioned in the ticket (break down of votes, slots in the voting period etc).

We seem to have a bit of a mix of 1 and 3 here where we're getting a state and then returning a subset of deposits.

Reply:
@ajsutton I was trying to implement first one. Do you mean that it includes the whole deposits map? including deposits that are already on BeaconChain and those that are not yet?

Reply:
Yes, the idea is to be able to see what we currently have in the deposit pool.

Reply:
Got it